### PR TITLE
chore(kit): `Range` refactor of `keySteps`

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [X] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Final PR for #256

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: projects/kit/utils/math/key-steps.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import {round} from '@taiga-ui/cdk';
+import {TUI_FLOATING_PRECISION} from '@taiga-ui/kit/constants';
+import {TuiKeySteps} from '@taiga-ui/kit/types';
+
+function tuiFindKeyStepsBoundariesByFn(
```
Comment (splincode) [inline projects/kit/utils/math/key-steps.ts:5]: is it worth write unit tests for them without considering component?

**Review by splincode — APPROVED** [review]

File: projects/kit/utils/math/key-steps.ts (hunk 1 above)
Comment (nsbarsukov) [reply projects/kit/utils/math/key-steps.ts:5]: We have already 92 unit tests + 12 cypress tests to test keySteps (Slider + Range) 🤔 

It is not difficult for me to repeat the same scenarios for functions from

https://github.com/Tinkoff/taiga-ui/blob/e7f3d1bfedf0da6eb38ac16ce6a0761838e373b1/projects/kit/components/slider/test/slider-key-steps.spec.ts#L61-L92 

Should I ?

Comment (splincode) [reply projects/kit/utils/math/key-steps.ts:5]: No, but as I understand it, the component still does some transformations

Comment (nsbarsukov) [reply projects/kit/utils/math/key-steps.ts:5]: > the component still does some transformations

Yes, they does.
I think I can come up some with a few new tests for this function.
For type coverage, at least))

Comment (nsbarsukov) [reply projects/kit/utils/math/key-steps.ts:5]: I added new test cases for this functions ✅

Comment (splincode) [reply projects/kit/utils/math/key-steps.ts:5]: amazing!

File: projects/kit/utils/math/key-steps.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    keySteps: TuiKeySteps,
+    fn: ([keyStepPercent, keyStepValue]: [number, number]) => boolean,
+): [[number, number], [number, number]] {
+    const keyStepUpperIndex = keySteps.findIndex((ketStep, i) => i && fn(ketStep));
```
Comment (waterplea) [inline projects/kit/utils/math/key-steps.ts:9]: `keyStep` - typo

**Review by waterplea — APPROVED** [review]

**Pull request merged**